### ISSUE TITLE
lr0 is ignored when optimizer is set to `auto`

### ISSUE THREAD

**Issue by Yoloex** [ultralytics/ultralytics#22782]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hi team, I'm trying to train YOLOv11x-cls with 3 classes using transfer learning.
Though I tried to set lr to small (e.g. 1e-4), the trainer ignored this one when checking logs on terminal.
It was saying as follows.
```
optimizer: `optimizer=auto` found, ignoring lr=0.004 and momentum=0.9 and determining best optimizer, lr and momentum automatically.
```

Can you lemme know if there is any configuration to use custom lr? Also, when I set optimizer=`Adam`, got an error saying NoneType object doesn't have `train` method.

### Environment

```bash
Ultralytics 8.3.228  Python-3.10.0 torch-2.5.0+cu124 CUDA:0 (NVIDIA GeForce RTX 4090, 24564MiB)
Setup complete  (32 CPUs, 95.7 GB RAM, 770.4/953.9 GB disk)

OS                     Windows-10-10.0.26200-SP0
Environment            Windows
Python                 3.10.0
Install                pip
Path                   C:\Users\user\miniconda3\envs\bitmind\Lib\site-packages\ultralytics
RAM                    95.72 GB
Disk                   770.4/953.9 GB
CPU                    Intel Core i9-14900KF
CPU count              32
GPU                    NVIDIA GeForce RTX 4090, 24564MiB
GPU count              1
CUDA                   12.4

matplotlib              3.10.7>=3.3.0
opencv-python           4.12.0.88>=4.6.0
pillow                  12.0.0>=7.1.2
pyyaml                  6.0.3>=5.3.1
requests                2.32.5>=2.23.0
scipy                   1.15.3>=1.4.1
torch                   2.5.0+cu124>=1.8.0
torch                   2.5.0+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision             0.20.0+cu124>=0.9.0
psutil                  7.1.3
polars                  1.35.2
ultralytics-thop        2.0.18>=2.0.18
{'OS': 'Windows-10-10.0.26200-SP0', 'Environment': 'Windows', 'Python': '3.10.0', 'Install': 'pip', 'Path': 'C:\\Users\\user\\miniconda3\\envs\\bitmind\\Lib\\site-packages\\ultralytics', 'RAM': '95.72 GB', 'Disk': '770.4/953.9 GB', 'CPU': 'Intel Core i9-14900KF', 'CPU count': 32, 'GPU': 'NVIDIA GeForce RTX 4090, 24564MiB', 'GPU count': 1, 'CUDA': '12.4', 'Package Info': {'numpy': '✅ 2.2.6>=1.23.0', 'matplotlib': '✅ 3.10.7>=3.3.0', 'opencv-python': '✅ 4.12.0.88>=4.6.0', 'pillow': '✅ 12.0.0>=7.1.2', 'pyyaml': '✅ 6.0.3>=5.3.1', 'requests': '✅ 2.32.5>=2.23.0', 'scipy': '✅ 1.15.3>=1.4.1', 'torch': '✅ 2.5.0+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.20.0+cu124>=0.9.0', 'psutil': '✅ 7.1.3', 'polars': '✅ 1.35.2', 'ultralytics-thop': '✅ 2.0.18>=2.0.18'}}
```

### Minimal Reproducible Example

```python
import os
import torch
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from torch.optim.lr_scheduler import OneCycleLR
from ultralytics import YOLO
from ultralytics.models.yolo.classify.train import ClassificationTrainer
from ultralytics.utils.callbacks import tensorboard as tb_cb

DEVICE = "cuda" if torch.cuda.is_available() else "cpu"

SAMPLE_0 = len(os.listdir("F:\\solution\\datasets\\custom\\train\\0"))
SAMPLE_1 = len(os.listdir("F:\\solution\\datasets\\custom\\train\\1"))
SAMPLE_2 = len(os.listdir("F:\\solution\\datasets\\custom\\train\\2"))

class CustomClassificationTrainer(ClassificationTrainer):
    def build_optimizer(self, model, name="auto", lr=None, momentum=None, decay=None, iterations=None):
        optimizer = super().build_optimizer(model, name=name, lr=lr, momentum=momentum, decay=decay, iterations=iterations)
        steps_per_epoch = (SAMPLE_0 + SAMPLE_1 + SAMPLE_2) // self.args.batch
        self.scheduler = OneCycleLR(
            optimizer,
            max_lr=self.args.lr0,
            epochs=self.args.epochs,
            steps_per_epoch=steps_per_epoch,
            pct_start=0.3,
            anneal_strategy="cos",
            cycle_momentum=True,
            base_momentum=0.85,
            max_momentum=0.95,
            div_factor=25.0,
            final_div_factor=10000.0
        )
        return optimizer
    

def on_fit_epoch_end(trainer):
    cm_obj = trainer.validator.confusion_matrix
    cm = getattr(cm_obj, "matrix", getattr(cm_obj, "confusion_matrix", None))
    if cm is None:
        return  # nothing to log yet
    if isinstance(cm, torch.Tensor):
        cm = cm.cpu().numpy()
    
    cm = cm[:-1, :-1].astype(np.int64)

    true_positives = np.diag(cm)
    actual_totals = cm.sum(axis=0)
    recall_per_class = true_positives / (actual_totals + 1e-8)
    step = trainer.epoch + 1

    fig, ax = plt.subplots(figsize=(8, 8))
    sns.heatmap(cm, annot=True, fmt='d', cmap='Blues', ax=ax)
    ax.set_xlabel("Actual")
    ax.set_ylabel("Predicted")
    ax.set_title(f"Confusion Matrix Epoch {step}")

    # write the figure to TensorBoard
    tb_cb.WRITER.add_figure("val/confusion_matrix", fig, global_step=step)
    plt.close(fig)

    for i, r in enumerate(recall_per_class):
        tb_cb._log_scalars({f"recall/class_{i}": float(r)}, step=step)

    tb_cb._log_scalars({"recall/mean": float(np.mean(recall_per_class))}, step=step)


model = YOLO("yolo11x-cls.pt")
model.add_callback("on_fit_epoch_end", on_fit_epoch_end)

if __name__ == "__main__":
    model.train(
        data="F:\\solution\\datasets\\custom",
        trainer=CustomClassificationTrainer,
        epochs=50,
        imgsz=224,
        batch=256,
        device=DEVICE,
        dropout=0.5,
        lr0=0.05,
        lrf=0.0001,
        cos_lr=False,
    )
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Yoloex, thank you for your detailed report and for using Ultralytics 🚀! This is an automated response to help you get unblocked as quickly as possible — an Ultralytics engineer will review your issue and assist you more specifically soon.

You can also check the [Docs](https://docs.ultralytics.com/) where you’ll find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that may clarify current optimizer and learning rate behaviors, as well as transfer learning configurations.

Since this appears to be a 🐛 Bug Report related to `optimizer=auto` and custom learning rate / optimizer settings, having a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) is very helpful for our team. Your snippet is already quite detailed 🎯 — if possible, please also include:

- Exact training command or script arguments used (including how you set `optimizer` and `lr0`)
- Full traceback for the `NoneType` error you mentioned when using `optimizer="Adam"`
- Any config file or overrides you might be using

If this evolves into a custom training ❓ Question (for example, about recommended optimizer or scheduler choices for your classification setup), please share as much context as you can, including dataset examples and logs, and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure your setup aligns with current recommendations.

Join the Ultralytics community wherever suits you best:
- For real-time chat and quick feedback, hop into [Discord](https://discord.com/invite/ultralytics) 🎧
- For more in-depth, long-form discussions, visit [Discourse](https://community.ultralytics.com/)
- To share results, tips, and discussion threads, check out our [Subreddit](https://www.reddit.com/r/Ultralytics/)  

## Upgrade

Before debugging further, please ensure you are on the latest `ultralytics` version with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) satisfied in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). This helps confirm whether the behavior is still present in the newest release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

These are good reference environments to test whether the `optimizer=auto` behavior and the `Adam` error reproduce consistently.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Once you’ve confirmed your version and, if possible, provided the full traceback for the `Adam` case, an Ultralytics engineer will be able to dig into this behavior in more detail ⚙️

**Comment by Y-T-G** [comment]
`optimizer="Adam"` should work. It's probably a bug in your custom trainer
